src/main/scala/org/ergoplatform/explorer/persistence/repositories/InputRepo.scala

```diff
@@ -0,0 +1,46 @@
+package org.ergoplatform.explorer.persistence.repositories
+
+import cats.data.NonEmptyList
+import cats.effect.Sync
+import cats.implicits._
+import doobie.implicits._
+import doobie.refined.implicits._
+import doobie.util.transactor.Transactor
+import org.ergoplatform.explorer.TxId
+import org.ergoplatform.explorer.persistence.models.Input
+import org.ergoplatform.explorer.persistence.models.composite.ExtendedInput
+import org.ergoplatform.explorer.persistence.doobieInstances._
+
+/** [[Input]] and [[ExtendedInput]] data access operations.
+  */
+trait InputRepo[F[_]] {
+
+  /** Put a given `input` to persistence.
+    */
+  def insert(input: Input): F[Unit]
+
+  /** Get all inputs related to a given `txId`.
+    */
+  def getAllByTxId(txId: TxId): F[List[ExtendedInput]]
```

Comment:
Let's avoid putting exccessive information into annotations.

Reply:
It doesn't matter in the sense of buisness-logic of an application. The result is only interpreted by the API user -- so we will describe it in API specification.

Reply:
Same for all queries results -- explorer doesnt interpret them somehow internally.

Reply:
Well, then please put in the code some hints about this. Where should I look in order to understand the code. Otherwise you will be recalled every time someone reads and tries to understand this public methods.

Reply:
Let's pull WTF ratio down.